LLVM string assert failure

## Context

Bill says LLVM 14 isn't supported, but the website says it is. Either way I'll post a bug report for you.

* Operating System & Odin Version:
* Please paste `odin report` output:
        Odin:    dev-2024-07:053f1759d
        OS:      Ubuntu 22.04.3 LTS, Linux 5.15.153.1-microsoft-standard-WSL2
        CPU:     Intel(R) Core(TM) i5-8350U CPU @ 1.70GHz
        RAM:     3856 MiB
        Backend: LLVM 14.0.0

## Expected Behavior

Compiled program

## Current Behavior

Error on build/run

## Failure Information (for bugs)

```
$ odin run main.odin -file
src/llvm_backend_const.cpp(179): Assertion Failure: `elem_type == LLVMTypeOf(values[i])` { [2 x %..string]*, i64 } != { %..string*, i64 }
Illegal instruction
```

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. `sudo apt get install clang llvm`
2. `git clone git@github.com:odin-lang/Odin`
3. `cd Odin`
4. `make` OR `./build_odin.sh`
5. Create file with string:
```
package main

import "core:fmt"

main :: proc() {
	fmt.println("Testing testing...")
}
```
6. `odin run main.odin -file`

I've also tried with LLVM 13 and LLVM 12 and get the same result.
I assume this error is from the tests that run when building Odin:

```
falconerd@DESKTOP-8U0CI5I:~/Odin$ make
./build_odin.sh debug
+ /usr/lib/llvm-12/bin/clang++ src/main.cpp src/libtommath.cpp -Wno-switch -Wno-macro-redefined -Wno-unused-value -DODIN_VERSION_RAW="dev-2024-07" -DGIT_SHA="053f1759d" -std=c++14 -I/usr/lib/llvm-12/include -std=c++14 -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -L/usr/lib/llvm-12/lib -g -pthread -lm -lstdc++ -ldl /usr/lib/llvm-12/lib/libLLVM-12.so -Wl,-rpath=$ORIGIN -o odin
+ set +x
src/llvm_backend_const.cpp(179): Assertion Failure: `elem_type == LLVMTypeOf(values[i])` { [3 x %..string]*, i64 } != { %..string*, i64 }
Illegal instruction
make: *** [Makefile:10: debug] Error 132
```

--- a/src/llvm_backend_const.cpp
+++ b/src/llvm_backend_const.cpp
@@ -176,7 +176,6 @@ gb_internal LLVMValueRef llvm_const_named_struct_internal(LLVMTypeRef t, LLVMVal
 	for (unsigned i = 0; i < elem_count; i++) {
 		LLVMTypeRef elem_type = LLVMStructGetTypeAtIndex(t, i);
 		values[i] = llvm_const_cast(values[i], elem_type);
-		GB_ASSERT_MSG(elem_type == LLVMTypeOf(values[i]), "%s != %s", LLVMPrintTypeToString(LLVMTypeOf(values[i])), LLVMPrintTypeToString(elem_type));
 	}
 	return LLVMConstNamedStruct(t, values, value_count);
 }

